emulator.c

```diff
@@ -154,8 +210,18 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       printf("ANI ");
       break;
     case 0xeb: // NOLINT
-      printf("XCHG");
-      break;
+      {        // XCHG
+        // exchange h and d
+        uint8_t temp = cpu->h;
```

Comment:
This is more just a comment. If I'm thinking about this correctly, this temporary variable goes on a stack outside of our emulator, right? I don't think that matters in terms, but I am curious how this is implemented on the actual hardware -- assuming it gets stored in one of the other registers.